src/useFetch.js

```diff
@@ -0,0 +1,34 @@
+// useFetch custom hook
+import { useState, useEffect } from "react";
+const useFetch = (url) => {
+    const [data, setData] = useState(null);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+    /*
```

Comment:
we are not passing an empty array now, but the url parameter instead. This is kept as a reminder of empty array effects